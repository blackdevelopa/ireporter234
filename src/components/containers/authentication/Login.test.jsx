import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import configMockStore from 'redux-mock-store';
// import { Form, Button } from 'semantic-ui-react';
import ConnectedLogin, { LoginForm } from './Login';
// import authReducer from '../../../redux/actions/auth/login';

const props = {
  loginUser: jest.fn(),
  history: {
    push: jest.fn(),
  },
  isAuthenticated: false,
};

const initialState = {
  isLoading: false,
  success: false,
  error: false,
  auth: {
    isAuthenticated: false,
  },
};

jest.mock('../../../redux/actions/auth/login');
jest.mock('../../../store/');
const mockStore = configMockStore([thunk]);
const store = mockStore(initialState);

const MyContext = React.createContext();

const wrapper = mount(
  <Provider store={store} context={MyContext}>
    <Router>
      <ConnectedLogin context={MyContext} />
    </Router>
  </Provider>
);

describe('<Login> ', () => {
  it('should render without login component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should test that form is submitted', () => {
    const loginForm = mount(<LoginForm {...props} />);
    const form = loginForm.find('Form');
    const spy = jest.spyOn(loginForm.instance(), 'handleSubmit');
    loginForm.instance().forceUpdate();
    form.simulate('submit');
    expect(spy).toBeCalled();
  });

  it('should render two form inputs', () => {
    const loginForm = mount(<LoginForm {...props} />);
    const form = loginForm.find('input');
    expect(form).toHaveLength(2);
  });

  it('should call handleOnchange', () => {
    const loginForm = mount(<LoginForm {...props} />);
    const spy = jest.spyOn(loginForm.instance(), 'onChange');
    const event = { target: { name: 'email', value: 'my@email.com' } };
    loginForm.instance().forceUpdate();
    loginForm.find('input[name="email"]').simulate('change', event);
    expect(spy).toBeCalled();
    expect(loginForm.instance().state.email).toBe('my@email.com');
  });

  it('should redirect to "/redflags if isAuthenticated is true"', () => {
    const loginForm = mount(<LoginForm {...props} />);
    loginForm.setProps({ isAuthenticated: true });
    expect(loginForm.props().history.push).toBeCalledWith('/red-flags');
  });
});
