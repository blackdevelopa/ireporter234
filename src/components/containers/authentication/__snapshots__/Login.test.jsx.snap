// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login />  should render without crashing 1`] = `
<Provider
  context={
    Object {
      "$$typeof": Symbol(react.context),
      "Consumer": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": [Circular],
      },
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": Object {},
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "81d863",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "81d863",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(withRouter(LoginForm))
        context={
          Object {
            "$$typeof": Symbol(react.context),
            "Consumer": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": [Circular],
            },
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": Object {},
            "_currentRenderer2": null,
            "_currentValue": undefined,
            "_currentValue2": undefined,
          }
        }
      >
        <withRouter(LoginForm)
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": undefined,
              "_currentValue2": undefined,
            }
          }
          isAuthenticated={false}
          loginUser={[Function]}
        >
          <Route>
            <LoginForm
              context={
                Object {
                  "$$typeof": Symbol(react.context),
                  "Consumer": Object {
                    "$$typeof": Symbol(react.context),
                    "_calculateChangedBits": null,
                    "_context": [Circular],
                  },
                  "Provider": Object {
                    "$$typeof": Symbol(react.provider),
                    "_context": [Circular],
                  },
                  "_calculateChangedBits": null,
                  "_currentRenderer": Object {},
                  "_currentRenderer2": null,
                  "_currentValue": undefined,
                  "_currentValue2": undefined,
                }
              }
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "81d863",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "81d863",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              isAuthenticated={false}
              location={
                Object {
                  "hash": "",
                  "key": "81d863",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              loginUser={[Function]}
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="sweet-loading"
              >
                <Form
                  as="form"
                  id="loginForm"
                  onSubmit={[Function]}
                >
                  <form
                    className="ui form"
                    id="loginForm"
                    onSubmit={[Function]}
                  >
                    <FormField
                      control={[Function]}
                      label="Email Address"
                      name="email"
                      onChange={[Function]}
                      placeholder="name@email.com"
                      required={true}
                      type="email"
                    >
                      <div
                        className="required field"
                      >
                        <label>
                          Email Address
                        </label>
                        <Input
                          name="email"
                          onChange={[Function]}
                          placeholder="name@email.com"
                          required={true}
                          type="email"
                        >
                          <div
                            className="ui input"
                          >
                            <input
                              name="email"
                              onChange={[Function]}
                              placeholder="name@email.com"
                              required={true}
                              type="email"
                            />
                          </div>
                        </Input>
                      </div>
                    </FormField>
                    <FormField
                      control={[Function]}
                      input="Password"
                      label="Password"
                      name="password"
                      onChange={[Function]}
                      placeholder="secret"
                      required={true}
                    >
                      <div
                        className="required field"
                      >
                        <label>
                          Password
                        </label>
                        <Input
                          input="Password"
                          name="password"
                          onChange={[Function]}
                          placeholder="secret"
                          required={true}
                          type="text"
                        >
                          <div
                            className="ui input"
                          >
                            <input
                              name="password"
                              onChange={[Function]}
                              placeholder="secret"
                              required={true}
                              type="Password"
                            />
                          </div>
                        </Input>
                      </div>
                    </FormField>
                    <Button
                      as="button"
                      style={
                        Object {
                          "background": "green",
                          "color": "white",
                        }
                      }
                      type="submit"
                    >
                      <Ref
                        innerRef={
                          Object {
                            "current": <button
                              class="ui button"
                              style="background: green; color: white;"
                              type="submit"
                            >
                              Login
                            </button>,
                          }
                        }
                      >
                        <RefFindNode
                          innerRef={
                            Object {
                              "current": <button
                                class="ui button"
                                style="background: green; color: white;"
                                type="submit"
                              >
                                Login
                              </button>,
                            }
                          }
                        >
                          <button
                            className="ui button"
                            onClick={[Function]}
                            style={
                              Object {
                                "background": "green",
                                "color": "white",
                              }
                            }
                            type="submit"
                          >
                            Login
                          </button>
                        </RefFindNode>
                      </Ref>
                    </Button>
                  </form>
                </Form>
              </div>
            </LoginForm>
          </Route>
        </withRouter(LoginForm)>
      </Connect(withRouter(LoginForm))>
    </Router>
  </MemoryRouter>
</Provider>
`;
